test/test_helpers.py

```diff
@@ -39,16 +39,20 @@ def cleanup():
 
     with (test_case.fake_script_path / name).open('wt') as f:
         # Before executing the provided script, dump the provided arguments in calls.txt.
+        # \x1e (\U001E) is "record separator" which seems like a reasonable one to use for this
+        # note we need something a little bit fancy to support multilines
+        # without having to worry about escaping things in the common case
+        # (if you want to look at the 'escaping' route, look at printf %q)
         f.write('''#!/bin/bash
-{ echo -n $(basename $0); printf ";%s" "$@"; echo; } >> $(dirname $0)/calls.txt
+{ echo -n $(basename $0); printf "\x1e%s" "$@"; echo; echo %%; echo; } >> $(dirname $0)/calls.txt
 ''' + content)
     os.chmod(str(test_case.fake_script_path / name), 0o755)
 
 
 def fake_script_calls(test_case, clear=False):
     try:
         with (test_case.fake_script_path / 'calls.txt').open('r+t') as f:
-            calls = [line.split(';') for line in f.read().splitlines()]
+            calls = [line.split('\x1e') for line in f.read().split('\n%%\n\n')[:-1]]
```

Comment:
It took me a long time to figure this out. What you're trying to do is tell that we made a call from the charm to 'juju-log' that itself wraps over multiple lines. Which might have any of lots of characters in them.
If you are going to the effort to use "\x1e" as the parameter separator, why not use something better as the end-of-text indicator that "\n%%\n\n" ?